java/org/apache/catalina/manager/host/HTMLHostManagerServlet.java

```diff
@@ -173,6 +177,36 @@ protected String add(HttpServletRequest request,String name,
         return stringWriter.toString();
     }
 
+    /**
+     * Log out by invalidating the current session and sending 401
+     * in order to prompt user for new login upon next access.
+     *
+     * @param request The Servlet request
+     * @param response The Servlet response
+     */
+    protected void logout(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setHeader("WWW-Authenticate","Basic realm=\"Login required\"");
```

Comment:
I don't like this because you assume from within the code that the auth mech is basic which is not always true.